samples/subsys/fs/littlefs_blk/README.rst

```diff
@@ -0,0 +1,57 @@
+.. _littlefs_blk-sample:
+
+littlefs File System Sample Application on block device
+#######################################################
+
+Overview
+********
+
+This sample app demonstrates use of Zephyr's :ref:`file system API
+<file_system_api>` over `littlefs`_, using a file system stored on
+the block device (i.e. SD card connected via SPI)
+
+Two tests are provided:
+
+1. The ``foo.txt`` file contains the value incremented by one at each
+boot of the board.
+
+2. The ``baz.txt`` file has a binary pattern, which spans on multiple
+sectors (512B each). Some binary values are adjusted to check
+if the pattern is not corrupted between booting the board.
+
+.. _littlefs:
+   https://github.com/ARMmbed/littlefs
+
+Requirements
+************
+
+One needs to prepare the SD card with littlefs file system on
+the HOST machine with the `lfs`_ program.
+
+.. _lfs:
+   https://www.thevtool.com/mounting-littlefs-on-linux-machine/
+
+.. code-block:: console
+
+   sudo chmod a+rw /dev/sda
+   lfs -d -s -f --read_size=512 --prog_size=512 --block_size=512 --cache_size=512 --lookahead_size=8192 --format /dev/sda
+   lfs -d -s -f --read_size=512 --prog_size=512 --block_size=512 --cache_size=512 --lookahead_size=8192 /dev/sda ./mnt_littlefs
+   cd ./mnt_littlefs
+   echo -en '\x01' > foo.txt
+   cd -
+   fusermount -u ./mnt_littlefs
+
+
+
+Building and Running
+********************
+
+This example is supposed to work on :ref:`Nucleo H743ZI <nucleo_h743zi_board>`
```

Comment:
:arrow_up:

Reply:
This example has been devised and initially tested on :ref:`Nucleo H743ZI <nucleo_h743zi_board>`
board. It can be also run on any other board with SD card connected to it.